homeassistant/components/calendar/__init__.py

```diff
@@ -194,18 +204,21 @@ def __init__(self, component: EntityComponent) -> None:
         """Initialize calendar view."""
         self.component = component
 
-    async def get(self, request, entity_id):
+    async def get(self, request: web.Request, entity_id: str) -> web.Response:
         """Return calendar events."""
         entity = self.component.get_entity(entity_id)
         start = request.query.get("start")
         end = request.query.get("end")
-        if None in (start, end, entity):
+        if start is None or end is None or entity is None:
             return web.Response(status=HTTPStatus.BAD_REQUEST)
+        assert isinstance(entity, CalendarEventDevice)
         try:
             start_date = dt.parse_datetime(start)
             end_date = dt.parse_datetime(end)
         except (ValueError, AttributeError):
             return web.Response(status=HTTPStatus.BAD_REQUEST)
+        assert start_date
```

Comment:
Should we return bad request instead if start_date or end_date is None?

Reply:
Sounds good. Fixed.